net/shadowsocks/src/opnsense/mvc/app/models/OPNsense/Shadowsocks/General.xml

```diff
@@ -46,5 +46,14 @@
                 <chacha20-ietf-poly1305>ChaCha20-IETF-Poly1305</chacha20-ietf-poly1305>
             </OptionValues>
         </cipher>
+        <tcpudpmode type="OptionField">
+            <default>tcp_only</default>
+            <Required>Y</Required>
+            <OptionValues>
+                <tcp_only>TCP only</tcp_only>
+                <udp_only>UDP only</udp_only>
+                <tcp_and_udp>Both</tcp_and_udp>
```

Comment:
Maybe call this "TCP and UDP". Otherwise it will say "Mode: Both" in the GUI, which could make it confusing at first glance.